metricbeat/module/kubernetes/util/kubernetes_test.go

```diff
@@ -592,6 +601,144 @@ func TestBuildMetadataEnricher_EventHandler(t *testing.T) {
 	resourceWatchers.lock.Unlock()
 }
 
+func TestBuildMetadataEnricher_PartialMetadata(t *testing.T) {
+	resourceWatchers := NewWatchers()
+
+	resourceWatchers.lock.Lock()
+	watcher := &metaWatcher{
+		watcher: &mockWatcher{
+			store: cache.NewStore(cache.MetaNamespaceKeyFunc),
+		},
+		started:         false,
+		metricsetsUsing: []string{"replicaset"},
+		enrichers:       make(map[string]*enricher),
+	}
+	resourceWatchers.metaWatchersMap[ReplicaSetResource] = watcher
+	addEventHandlerToWatcher(watcher, resourceWatchers)
+	resourceWatchers.lock.Unlock()
+
+	isController := true
+	resource := &metav1.PartialObjectMetadata{
+		ObjectMeta: metav1.ObjectMeta{
+			UID:  types.UID("mockuid"),
+			Name: "enrich",
+			Labels: map[string]string{
+				"label": "value",
+			},
+			Namespace: "default",
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					APIVersion: "apps/v1",
+					Kind:       "Deployment",
+					Name:       "enrich_deployment",
+					Controller: &isController,
+				},
+			},
+		},
+	}
+
+	config := &kubernetesConfig{
+		Namespace:  "test-ns",
+		SyncPeriod: time.Minute,
+		Node:       "test-node",
+		AddResourceMetadata: &metadata.AddResourceMetadataConfig{
+			CronJob:    false,
+			Deployment: true,
+		},
+	}
+
+	metricset := "replicaset"
+	log := logp.NewLogger(selector)
+
+	commonMetaConfig := metadata.Config{}
+	commonConfig, _ := conf.NewConfigFrom(&commonMetaConfig)
+	client := k8sfake.NewSimpleClientset()
+	generalMetaGen := metadata.NewResourceMetadataGenerator(commonConfig, client)
+
+	updateFunc := getEventMetadataFunc(log, generalMetaGen, nil, nil)
+
+	deleteFunc := func(r kubernetes.Resource) []string {
```

Comment:
Why we need the deleteFunc here in the test?

Reply:
We don't, good catch.

Reply:
We are testing the delete trigger, so we need some function there. I copied the one we actually use in `NewResourceMetadataEnricher`.